aspnet/fundamentals/static-files.rst

```diff
@@ -185,110 +144,60 @@ As with the ``UseStaticFiles``, ``UseDefaultFiles``, and ``UseDirectoryBrowser``
   - test.png
   - default.html
 
-Using the hierarchy example above, you might want to enable static files, default files, and browsing for the ``MyStaticFiles`` directory. In the following code snippet, that is accomplished with a single call to ``UseFileServer``.
-
-.. code-block:: c#
-
-  // Enable all static file middleware (serving of static files, default files,
-  // and directory browsing) for the MyStaticFiles directory.
-  app.UseFileServer(new FileServerOptions()
-  {
-      FileProvider = new PhysicalFileProvider(@"D:\Source\WebApplication1\src\WebApplication1\MyStaticFiles"),
-      RequestPath = new PathString("/StaticFiles"),
-      EnableDirectoryBrowsing = true
-  });
-
-Using the example hierarchy and code snippet from above, here's what happens if the user browses to various URIs:
-
-  - ``http://<yourApp>/StaticFiles/test.png`` - The ``MyStaticFiles/test.png`` file will be served to and presented by the browser.
-  - ``http://<yourApp>/StaticFiles`` - Since a default file is present (``MyStaticFiles/default.html``), that file will be served. If that file didn't exist, the browser would present a list of files in the ``MyStaticFiles`` directory (because the ``FileServerOptions.EnableDirectoryBrowsing`` property is set to ``true``).
-
-Working with content types
---------------------------
-
-The ASP.NET static files middleware understands almost 400 known file content types. If the user attempts to reach a file of an unknown file type, the static file middleware will not attempt to serve the file.
-
-Let's take the following directory/file hierarchy example to illustrate:
-
-- wwwroot
-
-  - css
-  - images
+Using the hierarchy example above, you might want to enable static files, default files, and browsing for the ``MyStaticFiles`` directory. In the following code snippet, that is accomplished with a single call to :dn:class:`~Microsoft.AspNetCore.Builder.FileServerOptions`.
 
-    - test.image
+.. literalinclude:: static-files/sample/StartupUseFS.cs
+  :language: c#
+  :lines: 55-61
+  :dedent: 12
 
-  - ...
+Using the file hierarchy and code above:
 
-Using this hierarchy, you could enable static file serving and directory browsing with the following:
+==========================================  ===================================
+URI                                         Response
+==========================================  ===================================
+\http://<app>/StaticFiles/test.png          StaticFiles/test.png
+\http://<app>/StaticFiles                   MyStaticFiles/default.html
```

Comment:
It's worth noting that DefaultFiles and DirectoryBrowser will take the url `http://<app>/StaticFiles` without the trailing slash will cause a client side redirect to `http://<app>/StaticFiles/` to add the slash. Otherwise relative urls within the documents would be incorrect.
https://github.com/aspnet/StaticFiles/blob/dev/src/Microsoft.AspNetCore.StaticFiles/DirectoryBrowserMiddleware.cs#L79-L84
https://github.com/aspnet/StaticFiles/blob/dev/src/Microsoft.AspNetCore.StaticFiles/DefaultFilesMiddleware.cs#L82-L87